src/data_generator/hdf5_generator.py

```diff
@@ -42,10 +42,10 @@ def generate(self):
         Generate hdf5 data for training. It generates a 3d dataset and writes it to file.
         """
         super().generate()
-        random.seed(10)
+        np.random.seed(10)
         samples_per_iter=1024
         dim1 = dim2 = self._dimension
```

Comment:
Are we missing variable dimensions here. Also this creation of dims and record is done per file in other generators.
This whole method can be abstracted in the parent class with only overriding how we store the files.